src/Transactions/TransferWithMemo.cs

```diff
@@ -18,27 +20,101 @@ public sealed record TransferWithMemo(CcdAmount Amount, AccountAddress Receiver,
     /// </summary>
     private const byte TransactionType = (byte)Types.TransactionType.TransferWithMemo;
 
+    /// <summary>
+    /// Prepares the account transaction payload for signing.
+    /// </summary>
+    /// <param name="sender">Address of the sender of the transaction.</param>
+    /// <param name="sequenceNumber">Account sequence number to use for the transaction.</param>
+    /// <param name="expiry">Expiration time of the transaction.</param>
+    public PreparedAccountTransaction Prepare(
+        AccountAddress sender,
+        AccountSequenceNumber sequenceNumber,
+        Expiry expiry
+    ) => new(sender, sequenceNumber, expiry, this._transactionCost, this);
+
+    /// <summary>
+    /// The transaction specific cost for submitting this type of
+    /// transaction to the chain.
+    ///
+    /// This should reflect the transaction-specific costs defined here:
+    /// https://github.com/Concordium/concordium-base/blob/78f557b8b8c94773a25e4f86a1a92bc323ea2e3d/haskell-src/Concordium/Cost.hs
+    /// </summary>
+    private readonly EnergyAmount _transactionCost = new(300);
+
+    /// <summary>
+    /// Gets the size (number of bytes) of the payload.
+    /// </summary>
+    internal override PayloadSize Size() => new(
+        this.Memo.SerializedLength() +
+        sizeof(TransactionType) +
+        CcdAmount.BytesLength +
+        AccountAddress.BytesLength);
+
+    /// <summary>
+    /// Create a "transfer with memo" payload from a serialized as bytes.
+    /// </summary>
+    /// <param name="bytes">The payload as bytes.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (TransferWithMemo? Transfer, string? Error) output)
+    {
+        var minSize = sizeof(TransactionType) + AccountAddress.BytesLength + CcdAmount.BytesLength + sizeof(ushort);
+        if (bytes.Length < minSize)
+        {
+            var msg = $"Invalid length in `TransferWithMemo.TryDeserial`. Expected at least {minSize}, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        };
+        if (bytes[0] != TransactionType)
+        {
+            var msg = $"Invalid transaction type in `Transfer.TryDeserial`. Expected {TransactionType}, found {bytes[0]}";
+            output = (null, msg);
+            return false;
+        };
+
+        var trxTypeLength = sizeof(TransactionType);
+        var accountLength = (int)AccountAddress.BytesLength;
+        var memoLength = BinaryPrimitives.ReadUInt16BigEndian(bytes[(trxTypeLength + accountLength)..]) + sizeof(ushort);
+
+        var accountBytes = bytes[trxTypeLength..];
```

Comment:
**Suggestion:** I have been trying to figure out what range indexing actually does for `Span`, but without luck, so just to be sure that we don't copy here, we should use the `.Slice` instead.

```suggestion
        var accountBytes = bytes.Slice(trxTypeLength);
```
Likewise for the other places using `bytes[x..]` in this function

Reply:
`Span` is just a contiguous region in memory so any slice would reference the same. Try
```
    [Fact]
    public void SpanSlicing()
    {
        // Arrange
        var someArray = new byte[] { 1, 2, 3, 4, 5, 6 };
        var someSpan = someArray.AsSpan();

        var someSlice = someSpan[..3];

        // Act
        someSlice[2] = 9; // Change value in slice affect initial array

        // Assert
        Assert.Equal(9, someArray[2]);
    }
```

Also if you use `Slice` the intellisense suggest you should use the range indexer.

Reply:
Actually, I have been using `Slice` but the formatter will change it so using `Slice` would give issues with it. I'd rather keep the range indexer.

Reply:
Great! Then ignore this suggestion.